src/functional/ffront/decorator.py

```diff
@@ -402,8 +413,11 @@ def program_inner(definition: types.FunctionType) -> Program:
     return program_inner if definition is None else program_inner(definition)
 
 
+OperatorNodeT = TypeVar("OperatorNodeT", bound=foast.LocatedNode)
+
+
 @dataclasses.dataclass(frozen=True)
-class FieldOperator(GTCallable):
+class FieldOperator(GTCallable, Generic[OperatorNodeT]):
```

Comment:
Just a comment: I find `FieldOperator[foast.FieldOperator]` a bit mind-bending. Would you consider to rename this `FieldOperator` class (and other `GTCallable` subtypes) to something like `FieldOpCallable` to make explicit the difference between both?

Reply:
Conclusion: We agree on the problem, but leave as is for now.